src/objective-c/RxLibrary/GRXBufferedPipe.m

```diff
@@ -33,11 +33,15 @@
 
 #import "GRXBufferedPipe.h"
 
+@interface GRXBufferedPipe ()
+@property(atomic) id<GRXWriteable> writeable;
+@property(atomic) BOOL inputIsFinished;
+@end
+
 @implementation GRXBufferedPipe {
-  id<GRXWriteable> _writeable;
-  NSMutableArray *_queue;
-  BOOL _inputIsFinished;
   NSError *_errorOrNil;
+  dispatch_queue_t _writeQueue;
+  dispatch_once_t _finishQueue;
```

Comment:
This is only used inside `writesFinishedWithError:`, which should only be called once already. So either it's unnecessary, or you're failing to protect something else with it.